pkg/lwip/include/arch/cc.h

```diff
@@ -58,7 +58,7 @@ extern "C" {
 #define S32_F   PRId32
 #define X32_F   PRIx32
 
-#define SZT_F   "lu"
+#define SZT_F   "zu"
```

Comment:
This is sadly not portable. With `nanospecs` for newlib (and I think also with MSP430) this will just print `zu` instead of the number. Also this seems not to be used (anymore?) so can this change be removed?

Reply:
The lwIP default is to make it PRIuPTR, which is probably better than setting it here.

SZT_F is used in a few lwip debug prints (api/sockets.c, core/mem.c)

Reply:
So this change can be removed?

Reply:
None of these defines are really required, since they are set to the same as upstream: https://git.savannah.nongnu.org/cgit/lwip.git/tree/src/include/lwip/arch.h?h=STABLE-2_1_x#n149

I can either change `SZT_F` to `PRIuPTR`, remove only `SZT_F`, or remove all these defines. Are they there to add documentation only?

Reply:
Mh if they are not used (anymore?) they can be removed. I was following the porting guide back then, that said those should be provided. However, this should be done in a separate PR.

Reply:
Changed it to `PRIuPTR` now, which is the minimum fix for this PR to work.

Reply:
I looked at the [porting docs](https://lwip.fandom.com/wiki/Porting_for_an_OS) now and indeed they say it should be defined, so lets keep them.